verilog/CST/port.cc

```diff
@@ -59,6 +59,11 @@ const SyntaxTreeLeaf* GetIdentifierFromModulePortDeclaration(
   return AutoUnwrapIdentifier(*ABSL_DIE_IF_NULL(identifier_symbol));
 }
 
+const SyntaxTreeLeaf* GetDirectionFromModulePortDeclaration(
+    const Symbol& symbol) {
+  return &verible::GetSubtreeAsLeaf(symbol, NodeEnum::kPortDeclaration, 0);
```

Comment:
This will fail where directions are omitted, since they are optional.

Recommend [CheckOptionalSymbolAsLeaf()](https://cs.opensource.google/verible/verible/+/master:common/text/tree_utils.h;bpv=1;bpt=1;l=156?q=GetSubtreeAsLeaf&ss=verible%2Fverible&gsn=CheckOptionalSymbolAsLeaf&gs=kythe%3A%2F%2Fgithub.com%2Fgoogle%2Fverible%3Flang%3Dc%252B%252B%3Fpath%3Dcommon%2Ftext%2Ftree_utils.h%23IHbB67ejQIuBHXFnjWcVINSNUElqhTFw5v_QsERjb3I).

Can you also add unit-test for this function, both with and without direction?

In port_test.cc, follow this test format:

https://cs.opensource.google/verible/verible/+/master:verilog/CST/port_test.cc;drc=fe94fa5b92ae668a3a6c4a6a05f9897cc0043fe3;l=300

using `TestVerilogSyntaxRangeMatches`.  This eliminates a ton of common boilerplate, and makes it extremely easy to check that matches correspond to precise ranges of text, and also reports unexpected results using a set comparison.

(I also would greatly appreciate it pre-existing tests in that file were migrated to this form, if you have the time.)